src/main/java/fr/sncf/osrd/train/TrainPhysicsIntegrator.java

```diff
@@ -165,6 +172,91 @@ private static double computeTimeDelta(double currentSpeed, double acceleration,
     }
 
+    /** Computes the next step of the integration method, based on an integrator and an action. */
+    public static PositionUpdate computeNextStep(TrainPhysicsIntegrator integrator,
+                                                 Action action,
+                                                 double distanceLeft,
+                                                 int directionSign) {
+
+        assert action != null;
+        assert action.type != Action.ActionType.EMERGENCY_BRAKING;
+        // run the physics sim
+        var update = integrator.computeUpdate(action, distanceLeft, directionSign);
+        return update;
+    }
+
+    /** Computes the next step of the integration method, based on location, speed, and a directive. */
+    public static PositionUpdate computeNextStep(TrainPositionTracker currentLocation,
+                                                 double currentSpeed,
+                                                 SpeedDirective directive,
+                                                 RollingStock rollingStock,
+                                                 double timeStep,
+                                                 double distanceLeft,
+                                                 int directionSign) {
+
+        var integrator = TrainPhysicsIntegrator.make(
+                timeStep,
+                rollingStock,
+                currentSpeed,
+                currentLocation.meanTrainGrade());
+        var action = integrator.actionToTargetSpeed(directive, rollingStock, directionSign);
+        return computeNextStep(
+                integrator,
+                action,
+                distanceLeft,
+                directionSign);
+    }
+
+    /** Computes the next step of the integration method, based on location, speed, and a specified action. */
+    public static PositionUpdate computeNextStep(TrainPositionTracker currentLocation,
+                                                 double currentSpeed,
+                                                 Action action,
+                                                 RollingStock rollingStock,
+                                                 double timeStep,
+                                                 double end,
+                                                 int directionSign) {
+
+        var integrator = TrainPhysicsIntegrator.make(
+                timeStep,
+                rollingStock,
+                currentSpeed,
+                currentLocation.meanTrainGrade());
+        var distanceLeft = end - currentLocation.getPathPosition();
+        return computeNextStep(
+                integrator,
+                action,
+                distanceLeft,
+                directionSign);
+    }
+
+    /** Computes the next step of the integration method, based on location, speed, and a set of controllers. */
+    public static PositionUpdate computeNextStep(TrainPositionTracker currentLocation,
```

Comment:
all the computeNextStep have slightly different uses, and yet the same name: I find it very hard to understand what all these are for
I think we should strive to find better specialized names

Reply:
They all do the same thing but with different input parameters, like computeUpdate or other methods in the code that are duplicated this way. As they all do the same thing I think it's a bit confusing to give them different names.